crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -56,8 +58,21 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                 return Ok(());
             };
 
-            // First entry has never any separator, doesn't matter which one we take.
-            write!(f, [first.format()])?;
+            if matches!(self.level, SuiteLevel::Nested)
+                && is_class_or_function_definition(first)
+                && !comments.has_leading_comments(first)
+            {
+                // Add an empty line for any nested functions or classes defined within non-function
+                // or class compound statements, e.g.:
+                // ```python
+                // if True:
```

Comment:
Nit: Should the example include the empty line? I was reading the example and it was unclear to me what empty line it would insert.